pkg/loader/load.go

```diff
@@ -624,6 +629,82 @@ func (b *batchManager) put(txn *Txn) error {
 	return nil
 }
 
+type txnManager struct {
+	cacheChan    chan *Txn
+	shutdown     chan struct{}
+	cachedSize   int
+	maxCacheSize int
+	cond         *sync.Cond
+	closed       bool
+}
+
+func newTxnManager(maxCacheSize int) *txnManager {
+	return &txnManager{
+		cacheChan:    make(chan *Txn, 1024),
+		maxCacheSize: maxCacheSize,
+		cond:         sync.NewCond(new(sync.Mutex)),
+		shutdown:     make(chan struct{}),
+	}
+}
+
+func (t *txnManager) put(input chan *Txn) chan *Txn {
+	ret := t.cacheChan
+	go func() {
+		for !t.closed {
+			var txn *Txn
+			var ok bool
+			select {
+			case txn, ok = <-input:
+				if !ok {
+					close(ret)
```

Comment:
only close `ret` when `input` is closed?

Reply:
Function `Close()` can close `ret` now.